hledger-lib/Hledger/Data/Dates.hs

```diff
@@ -751,12 +771,49 @@ reportinginterval = choice' [
                           thsuffix
                           many spacenonewline
                           string "day"
+                          many spacenonewline
                           optional $ do
                             many spacenonewline
                             string "of"
                             many spacenonewline
                             string "month"
-                          return $ DayOfMonth n
+                          return $ DayOfMonth n,
+                       do string "every"
+                          many spacenonewline
+                          n <- fmap read $ some digitChar
+                          thsuffix
+                          many spacenonewline
+                          ("",m,"") <- choice' [month, mon]
+                          many spacenonewline
+                          optional $ do
+                            many spacenonewline
+                            string "of"
+                            many spacenonewline
+                            string "year"
+                          return $ DayOfYear (read m) n,
+                       do string "every"
+                          many spacenonewline
+                          ("",m,"") <- choice' [month, mon]
+                          many spacenonewline
+                          n <- fmap read $ some digitChar
```

Comment:
I'm curious if it is possible to simplify this 3 cases with something like

```
do string "every"
  ...
  (m, d) <- choice' [
    ...
    ]
```

Maybe there is permutation/unordered combinator for list of parsers...

Reply:
I'v been looking for one as well, and failed. But we can factor out optional "of smthng" tail out of 4 parsers, and it will be less verbose...